sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/metrics/MeterMetric.java

```diff
@@ -0,0 +1,139 @@
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.implement.service.metrics;
+
+import com.google.common.collect.Lists;
+
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.DistributionSummary.Builder;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.config.MeterFilter;
+import io.sermant.core.common.BootArgsIndexer;
+import io.sermant.core.service.metric.Counter;
+import io.sermant.core.service.metric.DistributionStatisticConfig;
+import io.sermant.core.service.metric.Gauge;
+import io.sermant.core.service.metric.Metric;
+import io.sermant.core.service.metric.Summary;
+import io.sermant.core.service.metric.Tags;
+import io.sermant.core.service.metric.Timer;
+import io.sermant.core.service.metric.config.MetricConfig;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * metric
+ *
+ * @author zwmagic
+ * @since 2024-08-19
+ */
+public class MeterMetric implements Metric {
+    private static final Iterable<Tag> EMPTY = Collections.emptyList();
+
+    private final MeterRegistry registry;
+
+    /**
+     * Constructor that initializes a MeterMetric instance
+     *
+     * @param registryProvider An instance of MeterRegistryProvider used for creating and managing metrics.
+     * @param metricConfig MetricConfig
+     */
+    public MeterMetric(MeterRegistryProvider registryProvider, MetricConfig metricConfig) {
+        this.registry = registryProvider.getRegistry();
+        MeterRegistry.Config config = this.registry.config();
+        config.commonTags(
+                "agent", "sermant",
+                "agent.app.name", BootArgsIndexer.getAppName()
+        );
+        config.meterFilter(MeterFilter.maximumAllowableMetrics(metricConfig.getMaxTimeSeries()));
+    }
+
+    @Override
+    public Counter counter(String metricName, Tags tags, String description) {
+        io.micrometer.core.instrument.Counter counter = io.micrometer.core.instrument.Counter.builder(metricName)
+                .tags(getMeterTags(tags))
+                .description(description)
+                .register(registry);
+        return new MeterCounter(counter);
+    }
+
+    @Override
+    public Gauge gauge(String metricName, Tags tags, String description) {
+        return new MeterGauge(registry, metricName, getMeterTags(tags), description);
```

Comment:
Different metrics are constructed in different ways. Some data such as the transferred tag and description of the metric is in the metermetric method, and some is in the metric construction method（Gauge）.